### PR TITLE
feat: add kill count to collection log notifications

### LINKED ISSUE #371: Kill count for new collection log entries
### Checklist

- [X] I've searched the [issues and pull requests](https://github.com/pajlads/Dinkplugin/issues?q=) for similar looking suggestions.
- [X] I've checked the `Unreleased` section of the [changelog](https://github.com/pajlads/DinkPlugin/blob/master/CHANGELOG.md) for newly added features that sound like my suggestion.

### Describe your Suggestion

If available and possible add the kill count and monster that dropped the new collection log entry.

### Reasoning

It would be nice to see the information since some items can be dropped by multiple bosses and to see if someone is getting spooned

### PR DESCRIPTION
**Pull request by iProdigy**
Follow-up from #392; Closes #371

image

### CODE REVIEW

File: src/main/java/dinkplugin/notifiers/data/CollectionNotificationData.java [hunk 1]
```diff
@@ -28,13 +31,32 @@ public class CollectionNotificationData extends NotificationData {
...
+        if (completedEntries != null && totalEntries != null) {
+            fields.add(
+                new Field("Completed Entries", Field.formatProgress(completedEntries, totalEntries))
+            );
+        }
+        if (dropperKillCount != null) {
+            assert dropperName != null && dropperType != null;
+            fields.add(
+                new Field(
+                    Drop.getAction(dropperType) + " Count: " + dropperName,
```
Comment (iProdigy) [inline src/main/java/dinkplugin/notifiers/data/CollectionNotificationData.java:55]: is this fine (i.e., `Kill Count: NPC name` as field title), or should this be split into two separate fields (source & kill count)

Comment (pajlada) [reply src/main/java/dinkplugin/notifiers/data/CollectionNotificationData.java:55]: I prefer it as two separate fields, without looking at it

**Review by pajlada — APPROVED** [review]
Code looks good to me :+1: Haven't tested it

File: src/main/java/dinkplugin/notifiers/data/CollectionNotificationData.java (hunk 1 above)
Comment (iProdigy) [reply src/main/java/dinkplugin/notifiers/data/CollectionNotificationData.java]: Done in https://github.com/pajlads/DinkPlugin/pull/424/commits/e97606c0ad85b59e833bb82919335e1f98d120bb

**Review by pajlada — APPROVED** [review]

**Comment by Felanbird** [comment]
testing for the previous PR:

obor ✔️ 
vorkath ✔️ 
easy clue ✔️
zulrah ✔️ 
hespori ✔️

```
   "extra":{
      "items":[],
      "source":"Zulrah",
      "category":"NPC",
      "killCount":1502
```
```
   "extra":{
      "items":[],
      "source":"Vorkath",
      "category":"NPC",
      "killCount":2003
   },
```

i did not get a collection log entry while doing this testing

**Comment by iProdigy** [comment]
Verified in-game for collection

image

Screenshot_20240216_235353

**Pull request merged**